File: easy/python/0035.py
class Solution:
    def searchInsert(self, nums: 'List[int]', target: 'int') -> 'int':
        for idx, i in enumerate(nums):
            if i == target:
                return idx
            if i > target:
                return idx
        return len(nums)
        # left = 0
        # right = len(nums) -1
        # index = -1
        # if left == right:
        #     if nums[0] >= target:
        #         return 0
        #     else:
        #         return 1
        # while left < right:
        #     m = int((right + left)/2)
        #     if nums[m] < target:
        #         left = m + 1
        #     elif nums[m] > target:
        #         right = m - 1
        #     else:
        #         if nums[m] != target:
        #             break
        #         else:
        #             index = m
        #             break
        # final_index = int((left+right)/2)
        # if index == -1:
        #     if nums[final_index] >= target:
        #         if final_index -1 <0:
        #             return 0
        #         return final_index
        #     elif nums[final_index] < target:
        #         return final_index+1
        # else:
        #     return index

File: easy/c/0035.c
int searchInsert(int* nums, int numsSize, int target){
    // for(int i = 0 ; i < numsSize; i++)if(nums[i] >= target)return i;
    // return numsSize;
    if(nums[0] > target) return 0;
    if(nums[numsSize-1] < target) return numsSize;
    int left = 0, right = numsSize;
    int m = 0;
    while(left <= right){
        m = (left + right) / 2;
        if(nums[m] > target){
            right = m-1;
        }else if(nums[m] < target){
            left = m+1;
        }else{
            return m;
        }
    }
    if(nums[m] > target)return m;
    else if(nums[m] < target)return m+1;
    return numsSize;
}